### PR TITLE
fix: should not throw error when pass null

### PR DESCRIPTION
**Pull request by popomore**

### CODE REVIEW

File: lib/ready.js [hunk 1]
```diff
@@ -80,6 +80,9 @@ class Ready extends EventEmitter {
     opt.name = name || cacheKey;
     const timer = setTimeout(() => this.emit('ready_timeout', opt.name), opt.timeout);
     const cb = once(err => {
+      if (err != null && !(err instanceof Error)) {
+        err = new Error(err);
```
Comment (popomore) [inline lib/ready.js:84]: improve the stack

Comment (shaoshuai0102) [reply lib/ready.js:84]: Once err is undefined, this will be broken.

Should be `err !== null && err !== undefined` instead.

Comment (popomore) [reply lib/ready.js:84]: `err != null` is equal to `err !== null && err !== undefined`

Comment (shaoshuai0102) [reply lib/ready.js:84]: oh yes. I got wrong.

Comment (shaoshuai0102) [reply lib/ready.js:84]: How about moving this logic to readyDone? So you can deal with error together at one place.

Comment (popomore) [reply lib/ready.js:84]: Because of the stack, so this way is better rather than after setImmdiate

**Comment by shaoshuai0102** [comment]
U r right. LGTM

发自我的 iPad

**Review by fengmk2 — APPROVED** [review]

**Comment by fengmk2** [comment]
Going to merge and I will release a patch version.

**Comment by fengmk2** [comment]
2.0.1

**Pull request merged**